File: 2454-largest-local-values-in-a-matrix/largest-local-values-in-a-matrix.py
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        n = len(grid)
        max_local = []
        for i in range(n-2):
            row = []
            for j in range(n-2):
                max_value = max(
                    grid[i][j], grid[i][j+1], grid[i][j+2],
                    grid[i+1][j], grid[i+1][j+1], grid[i+1][j+2],
                    grid[i+2][j], grid[i+2][j+1], grid[i+2][j+2]
                )
                row.append(max_value)
            max_local.append(row)
        return max_local

File: 2454-largest-local-values-in-a-matrix/largest-local-values-in-a-matrix.js
/**
 * @param {number[][]} grid
 * @return {number[][]}
 */
var largestLocal = function(grid) {
    const n = grid.length;
    const ans = Array(n - 2).fill(0).map(()=>Array(n - 2).fill(0));
    for (let i = 0; i < (n-2); i++){
        for (let j = 0; j < (n-2); j++){
            max_num = Math.max(grid[i][j], grid[i][j+1], grid[i][j+2], grid[i+1][j], grid[i+1][j+1], grid[i+1][j+2], grid[i+2][j], grid[i+2][j+1], grid[i+2][j+2])
            ans[i][j] = max_num;
        }
    }
    return ans
};